File: OOP_Codes/OOP_17/BookShop/book.cpp
#include "book.h"

Book::Book(std::string a,std::string t,double pr):author(a),title(t),price(pr) {}
std::string Book::get_author()const {return this->author;}
std::string Book::get_title()const {return this->title;}
double Book::get_price()const {return this->price;}
QString Book::get_author_qt()
{
    QString auth="";
    int size=author.length();
    for(int i=0;i<size;i++)
    {
        auth+=author.at(i);
    }
    return auth;
}

QString Book::get_title_qt()
{
    QString tit="";
    int size=title.length();
    for(int i=0;i<size;i++)
    {
        tit+=title.at(i);
    }
    return tit;
}

File: JAVA/BOOK_JAVA/src/bookstore/Book.java
package bookstore;

import javax.swing.*;

public class Book {

  private String title, author;
  private int price;

  public Book(String t, String a, int p) {
    setTitle(t);
    setAuthor(a);
    setPrice(p);
  }

  public void setTitle(String t) {
    title = t;
  }

  public String getTitle() {
    return title;
  }

  public void setAuthor(String a) {
    author = a;
  }

  public String getAuthor() {
    return author;
  }

  public void setPrice(int p) {
    price = p;
  }

  public int getPrice() {
    return price;
  }

  @Override
  public String toString() {
    return title + ":" + author + ":" + price;
  }
}
